samples/cellular/battery/src/main.c

```diff
@@ -424,6 +424,11 @@ int main(void)
 
 		k_sleep(K_SECONDS(refresh_interval));
 	}
+	
+	err = nrf_modem_lib_shutdown();
```

Comment:
This is unreachable per now. The return statements (on error) in above while loop should shut down the modem library. Maybe add a `clean_up` goto as in the ciphersuites sample?